lms/services/jstor.py

```diff
@@ -22,8 +24,45 @@ class JSTORMetadataSchema(RequestsResponseSchema):
     title = fields.Str()
     subtitle = fields.Str()
 
-    # List of titles of works that this item is a review of.
     reviewed_works = fields.List(fields.Str())
+    """List of titles of works that this item is a review of."""
+
+    has_pdf = fields.Boolean(required=True)
+    """
+    Does this item have a PDF?
+
+    This can be false if the item is a collection (eg. of book chapters or
+    journal articles).
+    """
+
+    requestor_access_level = fields.Str(required=True)
+    """
+    Does the current institution have access to the PDF?
+
+    The "current" institution is the one identified by the site code specified
+    in the Authorization header of the request.
+
+    This will be "full_access" if the institution has access to the PDF,
+    or another value (eg. "preview_access") otherwise.
+    """
+
+
+class ContentStatus(Enum):
+    """Indicates whether an item has content available for use with Hypothesis."""
+
+    AVAILABLE = auto()
+    """Content is available for the item."""
+
+    NO_CONTENT = auto()
+    """Item does not have associated content (eg. a PDF)."""
+
+    NO_ACCESS = auto()
+    """This item has content, but the current institution does not have access to it."""
+
+
+class MetadataResult(TypedDict):
+    title: str
+    content_status: ContentStatus
```

Comment:
I think I'd probably skip this? It's nice documentation in one way, but it seems like a real bother to have to introduce a class per method which returns a dict, and to keep both in sync all the time. I think we might be better to embrace the chaos and accept Python's less typey nature.

There are some suggestions later on which suggest the API view would become a very thin wrapper and return whatever it's given, at which point we'd be advertising an interface nobody would really use (in Python).